### PR TITLE
Migrating to Spring HATEOAS 1.x

### PR DESCRIPTION
**Pull request by szvasas**
The project was upgraded to Spring Boot 2.2.1.RELEASE which includes Spring HATEOAS 1.0.1.RELEASE which brings some breaking changes causing the project not to compile.

The aim of my PR is to migrate to the new API and fix these compilation errors.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@szvasas Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-guides/gs-rest-hateoas&pullRequestId=25)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-guides/gs-rest-hateoas&pullRequestId=25) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@szvasas Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-guides/gs-rest-hateoas&pullRequestId=25)!

File: complete/src/main/java/hello/Greeting.java [hunk 1]
```diff
@@ -1,11 +1,11 @@
 package hello;
 
-import org.springframework.hateoas.ResourceSupport;
+import org.springframework.hateoas.RepresentationModel;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public class Greeting extends ResourceSupport {
+public class Greeting extends RepresentationModel {
```
Comment (mikewoudenberg) [inline complete/src/main/java/hello/Greeting.java]: `RepresentationModel` is a raw type. References to generic type RepresentationModel<T> should be parameterized.

I parameterized it by using `RepresentationModel<Greeting>`

Comment (szvasas) [reply complete/src/main/java/hello/Greeting.java]: Nice catch, thank you, I have fixed it.

**Comment by szvasas** [comment]
Hi @gregturn, @dsyer,

If you have some free cycles, can you please help me getting this fix committed?

Thanks,
Szabolcs

**Comment by Buzzardo** [comment]
Thank you for pointing out the fix for it. I included your fix in another fix, so I can close this PR. Thanks again.

**Pull request closed without merging**